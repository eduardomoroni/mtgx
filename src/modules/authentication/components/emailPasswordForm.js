// @flow

import React, { Component } from 'react'
import { PropTypes } from 'prop-types'
import I18n from 'react-native-i18n'
import { View } from 'react-native'
import { Input, Button } from 'nachos-ui'
import { styles } from './styles/emailPasswordForm.style'

export class EmailPasswordForm extends Component {
  state = {
    email: '',
    password: ''
  }

  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonCallback: PropTypes.func.isRequired
  }

  render () {
    const { buttonCallback, buttonText } = this.props
    const { email, password } = this.state

    return (
      <View>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Input
          icon='ios-lock-outline'
          style={[styles.input]}
          placeholder={I18n.t('PASSWORD')}
          value={this.state.password}
          onChangeText={password => this.setState({password})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={() => buttonCallback(email, password)}
        >
          {buttonText}
        </Button>
      </View>
    )
  }
}
