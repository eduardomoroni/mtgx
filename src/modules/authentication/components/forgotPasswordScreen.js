// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1, Button, Input } from 'nachos-ui'
import { styles } from './styles/forgotPassword.style'

export class ForgotPasswordScreen extends Component {
  state = {
    email: ''
  }

  static propTypes = {
    resetPassword: PropTypes.func.isRequired
  }

  onForgotPasswordPressed = async () => {
    this.props.resetPassword(this.state.email)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('RESET_PASSWORD')}</H1>
        <Input
          icon='ios-mail-outline'
          style={[styles.input]}
          placeholder={I18n.t('EMAIL')}
          value={this.state.email}
          onChangeText={email => this.setState({email})}
        />
        <Button
          kind='squared'
          type='success'
          style={[styles.button]}
          onPress={this.onForgotPasswordPressed}
        >
          {I18n.t('RESET_PASSWORD')}
        </Button>
      </View>
    )
  }
}
