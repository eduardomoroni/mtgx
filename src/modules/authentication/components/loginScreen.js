// @flow

import React, { Component } from 'react'
import Config from 'react-native-config'
import I18n from 'react-native-i18n'
import { StyleSheet, Text, View } from 'react-native'
import { H4, Button } from 'nachos-ui'

export class LoginScreen extends Component {
  renderButton () {
    const btnStyle = { margin: 15 }
    return (
      <View>
        <H4>Example:</H4>
        <Button kind='squared' type='success' style={btnStyle}>
          Success
        </Button>
        <Button kind='squared' type='danger' style={btnStyle}>
          Danger
        </Button>
        <Button
          kind='squared'
          iconName='md-cloud-download'
          style={btnStyle}
        >
          Primary
        </Button>
        <H4 align='center'>Disabled style</H4>
      </View>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <Text style={styles.instructions}>
          To get started, edit index.ios.js
        </Text>
        <Text style={styles.instructions}>
          Press Cmd+R to reload,{'\n'}
          Cmd+D or shake for dev menu
        </Text>
        <Text style={styles.instructions}>
          {Config.TEST_ENV}
        </Text>
        <Text style={styles.instructions}>
          {I18n.t('test')}
        </Text>
        {this.renderButton()}
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF'
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5
  }
})
