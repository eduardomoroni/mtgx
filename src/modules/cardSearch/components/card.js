// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { View, Text } from 'react-native'

import { placeholdersToSymbols } from '../../../services/realm/conversion/placeholder' // TODO: REVISIT
import { cardType, printingsType } from '../types/cardType'
import { styles } from './styles/card.styles'

// TODO: REVISIT
export const getLastPrinting = (printings: printingsType) => {
  const keys = Object.keys(printings)
  const lastPrinting = printings[keys[keys.length - 1]]
  return lastPrinting.printing
}

export class Card extends Component {
  static propTypes = {
    showCardText: PropTypes.bool,
    card: cardType
  }

  renderCardNameAndMana = () => {
    const { name, manaCost } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardName' style={styles.cardNameText}>
            {name}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardManaCost' style={styles.mana}>
            {placeholdersToSymbols(manaCost)}
          </Text>
        </View>
      </View>
    )
  }

  renderCardTypeAndEdition = () => {
    const { type, printings } = this.props.card

    return (
      <View style={styles.lineContainer}>
        <View style={styles.leftWord}>
          <Text key='cardType' style={styles.cardTypeText}>
            {type}
          </Text>
        </View>
        <View style={styles.rightWord}>
          <Text key='cardEdition'>
            {getLastPrinting(printings)}
          </Text>
        </View>
      </View>
    )
  }

  renderTextAndPower = () => {
    const { card, showCardText } = this.props
    const { text, power, toughness } = card

    return (
      <View style={styles.cardTextContainer}>
        <View style={styles.cardText}>
          <Text key='cardText' style={styles.text}>
            {showCardText ? text : ''}
          </Text>
        </View>
        <View style={styles.cardPower}>
          <Text key='cardPowerAndToughness' style={styles.cardPowerToughness}>
            {power || toughness ? `${power}/${toughness}` : ''}
          </Text>
        </View>
      </View>
    )
  }

  render () {
    return (
      <View style={styles.container}>
        {this.renderCardNameAndMana()}
        {this.renderCardTypeAndEdition()}
        {this.renderTextAndPower()}
      </View>
    )
  }
}
