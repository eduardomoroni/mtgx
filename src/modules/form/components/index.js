// @flow

export * from './inputLabel'
export * from './inputPicker'
export * from './textInputForm'
export * from './switchInputForm'
export * from './modalToggle'
export * from './multiSelect'
// $FlowFixMe
export * from './numericInputForm'
// $FlowFixMe
export * from './dropdownInputForm'
