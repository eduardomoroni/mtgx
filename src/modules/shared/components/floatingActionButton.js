// @flow

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import ActionButton from 'react-native-action-button'
import Icon from 'react-native-vector-icons/Ionicons'

import { styles } from './styles/floatingActionButton.styles'

const ItemType = {
  name: PropTypes.string.isRequired,
  buttonColor: PropTypes.string,
  title: PropTypes.string,
  onPress: PropTypes.func
}

export class FloatingActionButton extends Component {
  static propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape(ItemType)).isRequired
  }

  renderItem (item: ItemType, key: number) {
    const { name, ...itemProps } = item

    return (
      <ActionButton.Item {...itemProps} key={key}>
        <Icon name={name} style={styles.actionButtonIcon} />
      </ActionButton.Item>
    )
  }

  render () {
    return (
      <ActionButton buttonColor='rgba(231,76,60,1)'>
        {this.props.items.map(this.renderItem)}
      </ActionButton>
    )
  }
}
