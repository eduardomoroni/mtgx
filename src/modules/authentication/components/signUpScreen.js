// @flow

import React, { Component } from 'react'
import I18n from 'react-native-i18n'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import { H1 } from 'nachos-ui'
import { styles } from './styles/loginScreen.style'
import { EmailPasswordForm } from './emailPasswordForm'

export class SignUpScreen extends Component {
  static propTypes = {
    signUpUser: PropTypes.func.isRequired
  }

  onSignUpPressed = async (email: string, password: string) => {
    this.props.signUpUser(email, password)
  }

  render () {
    return (
      <View style={styles.container}>
        <H1>{I18n.t('SIGNUP')}</H1>
        <EmailPasswordForm
          buttonText={I18n.t('SIGNUP')}
          buttonCallback={this.onSignUpPressed}
        />
      </View>
    )
  }
}
