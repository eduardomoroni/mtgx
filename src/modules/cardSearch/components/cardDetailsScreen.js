// @flow

import I18n from 'react-native-i18n'
import React, { Component } from 'react'
import { Text, View, ScrollView } from 'react-native'

import { styles } from './styles/cardDetailsScreen.styles'
import { CardImage } from './cardImage'
import { cardType } from '../types/cardType'
import { FloatingActionButton } from '../../shared/components/floatingActionButton'

const fabItems = [
  { name: 'md-share',
    buttonColor: '#9b59b6',
    title: 'Share',
    onPress: () => null
  }, {
    name: 'md-folder-open',
    buttonColor: '#3498db',
    title: 'Add to Deck',
    onPress: () => null
  }, {
    name: 'md-swap',
    buttonColor: '#1abc9c',
    title: 'Add to TradeList',
    onPress: () => null
  }
]

export class CardDetailsScreen extends Component {
  static propTypes = {
    card: cardType
  }

  renderRow = (rowName: string, rowText: string) => {
    if (rowText === null) {
      return null
    }

    return (
      <View style={styles.rowContainer}>
        <Text style={styles.rowText}>
          <Text style={styles.rowName}>{`${I18n.t(rowName)}: `}</Text>
          {rowText}
        </Text>
      </View>
    )
  }

  render () {
    const { card } = this.props

    return (
      <ScrollView style={styles.container}>
        <CardImage style={styles.card} multiverseId={card.multiverseid} />
        {this.renderRow('CARD_NAME', card.name)}
        {this.renderRow('TYPE', card.type)}
        {this.renderRow('POWER_TOUGHNESS', `${card.power}/${card.toughness}`)}
        {this.renderRow('MANA_COST', `${card.manaCost} (${card.cmc})`)}
        {this.renderRow('RARITY', card.rarity)}
        {this.renderRow('CARD_TEXT', card.text)}
        {this.renderRow('FLAVOR_TEXT', card.flavor)}
        <FloatingActionButton items={fabItems} />
      </ScrollView>
    )
  }
}
