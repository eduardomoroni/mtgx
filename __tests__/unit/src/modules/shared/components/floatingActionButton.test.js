// @flow

import React from 'react'
import { shallow } from 'enzyme'
import ActionButton from 'react-native-action-button'
import { FloatingActionButton } from '../../../../../../src/modules/shared/components/floatingActionButton'

const props = {
  items: [
    { name: 'md-share',
      buttonColor: '#9b59b6',
      title: 'Share',
      onPress: jest.fn()
    }, {
      name: 'md-folder-open',
      buttonColor: '#3498db',
      title: 'Add to Deck',
      onPress: jest.fn()
    }, {
      name: 'md-swap',
      buttonColor: '#1abc9c',
      title: 'Add to TradeList',
      onPress: jest.fn()
    }
  ]
}

describe('<FloatingActionButton />', () => {
  const wrapper = shallow(<FloatingActionButton {...props} />)

  it('Should render InputLabel Component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should trigger callback on pressing button', () => {
    const button = wrapper.find(ActionButton.Item).at(0)
    button.simulate('press')
    expect(props.items[0].onPress).toHaveBeenCalled()
  })
})
