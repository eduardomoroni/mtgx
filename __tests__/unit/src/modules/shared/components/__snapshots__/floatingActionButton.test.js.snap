// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FloatingActionButton /> Should render InputLabel Component 1`] = `
<ActionButton
  active={false}
  activeOpacity={0.85}
  autoInactive={true}
  backdrop={false}
  backgroundTappable={false}
  bgColor="transparent"
  bgOpacity={1}
  buttonColor="rgba(231,76,60,1)"
  buttonText="+"
  buttonTextStyle={Object {}}
  degrees={45}
  fixNativeFeedbackRadius={false}
  nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
  offsetX={30}
  offsetY={30}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOn={[Function]}
  outRangeScale={1}
  position="right"
  resetToken={null}
  size={56}
  spacing={20}
  useNativeFeedback={true}
  verticalOrientation="up"
>
  <ActionButtonItem
    active={true}
    activeOpacity={0.85}
    buttonColor="#9b59b6"
    fixNativeFeedbackRadius={false}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    onPress={[Function]}
    spaceBetween={15}
    title="Share"
    useNativeFeedback={true}
  >
    <Icon
      allowFontScaling={false}
      name="md-share"
      size={12}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "height": 22,
        }
      }
    />
  </ActionButtonItem>
  <ActionButtonItem
    active={true}
    activeOpacity={0.85}
    buttonColor="#3498db"
    fixNativeFeedbackRadius={false}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    onPress={[Function]}
    spaceBetween={15}
    title="Add to Deck"
    useNativeFeedback={true}
  >
    <Icon
      allowFontScaling={false}
      name="md-folder-open"
      size={12}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "height": 22,
        }
      }
    />
  </ActionButtonItem>
  <ActionButtonItem
    active={true}
    activeOpacity={0.85}
    buttonColor="#1abc9c"
    fixNativeFeedbackRadius={false}
    nativeFeedbackRippleColor="rgba(255,255,255,0.75)"
    onPress={[Function]}
    spaceBetween={15}
    title="Add to TradeList"
    useNativeFeedback={true}
  >
    <Icon
      allowFontScaling={false}
      name="md-swap"
      size={12}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "height": 22,
        }
      }
    />
  </ActionButtonItem>
</ActionButton>
`;
